parsl/dataflow/errors.py

```diff
@@ -45,8 +47,9 @@ def __init__(self, dependent_exceptions_tids: Sequence[Tuple[Exception, str]], t
         self.task_id = task_id
 
     def __str__(self) -> str:
-        dep_tids = [tid for (exception, tid) in self.dependent_exceptions_tids]
-        return "Dependency failure for task {} with failed dependencies from tasks {}".format(self.task_id, dep_tids)
+        dep_ids = [tid for (exception, tid) in self.dependent_exceptions_tids]
+        deps = ", ".join(dep_ids)
+        return f"Dependency failure for task {self.task_id} with failed dependencies from {deps}"
```

Comment:
- Aren't using `exception`, so stylistically consider the underscore prefix
- Not that "it matters," but that could be a generator instead; no need for the interim data structure:
    ```python
    dep_ids = (tid for _exc, tid in self.dependent_exception_tids)
    deps = ", ".join()
    ```
    or just:
    ```python
    deps = ", ".join(tid for _exc, tid in self.dependent_exception_tids)
    ```